src/alchemlyb/tests/test_workflow_ABFE.py

```diff
@@ -0,0 +1,333 @@
+                        overlap='O_MBAR.pdf',
+                        breakdown=True, forwrev=10, log='result.log')
+        return workflow
+
+    def test_read(self, workflow):
+        '''test if the files has been loaded correctly.'''
+        assert len(workflow.u_nk_list) == 30
+        assert len(workflow.dHdl_list) == 30
+        assert all([len(u_nk) == 1001 for u_nk in workflow.u_nk_list])
+        assert all([len(dHdl) == 1001 for dHdl in workflow.dHdl_list])
+
+    def test_subsample(self, workflow):
+        '''Test if the data has been shrinked by subsampling.'''
+        assert len(workflow.u_nk_sample_list) == 30
+        assert len(workflow.dHdl_sample_list) == 30
+        assert all([len(u_nk) < 1001 for u_nk in workflow.u_nk_sample_list])
+        assert all([len(dHdl) < 1001 for dHdl in workflow.dHdl_sample_list])
+
+    def test_estomator(self, workflow):
+        '''Test if all three estimator has been used.'''
+        assert len(workflow.estimator) == 3
+        assert 'mbar' in workflow.estimator
+        assert 'ti' in workflow.estimator
+        assert 'bar' in workflow.estimator
+
+    def test_summary(self, workflow):
+        '''Test if if the summary is right.'''
+        summary = workflow.generate_result()
+        assert np.isclose(summary['MBAR']['Stages']['TOTAL'], 21.788, 0.1)
+
+    def test_O_MBAR(self, workflow):
+        '''test if the O_MBAR.pdf has been plotted.'''
+        assert os.path.isfile('O_MBAR.pdf')
+        os.remove('O_MBAR.pdf')
+
+    def test_dhdl_TI(self, workflow):
+        '''test if the dhdl_TI.pdf has been plotted.'''
+        assert os.path.isfile('dhdl_TI.pdf')
+        os.remove('dhdl_TI.pdf')
+
+    def test_dF_state(self, workflow):
+        '''test if the dF_state.pdf has been plotted.'''
+        assert os.path.isfile('dF_state.pdf')
+        os.remove('dF_state.pdf')
+        assert os.path.isfile('dF_state_long.pdf')
+        os.remove('dF_state_long.pdf')
+
+    def test_convergence(self, workflow):
+        '''test if the dF_state.pdf has been plotted.'''
+        assert os.path.isfile('dF_t.pdf')
+        os.remove('dF_t.pdf')
+        assert len(workflow.convergence) == 10
+
+class Test_manual_ABFE():
+    '''Test the manual workflow for load_ABFE from alchemtest.gmx for three
+    stage transformation.'''
+
+    @staticmethod
+    @pytest.fixture(scope='class')
+    def workflow():
+        dir = os.path.dirname(load_ABFE()['data']['complex'][0])
+        workflow = ABFE(software='Gromacs', dir=dir, prefix='dhdl',
+                        suffix='xvg', T=310)
+        workflow.update_units('kcal/mol')
+        workflow.preprocess(skiptime=10, uncorr='dhdl', threshold=50)
+        workflow.estimate(methods=('mbar', 'bar', 'ti'))
+        workflow.plot_overlap_matrix(overlap='O_MBAR.pdf')
+        workflow.plot_ti_dhdl(dhdl_TI='dhdl_TI.pdf')
+        workflow.plot_dF_state(dF_state='dF_state.pdf')
+        workflow.check_convergence(10, dF_t='dF_t.pdf')
```

Comment:
These could be run in the test methods when needed.

Reply:
@orbeckst So the aim of this fixture is to mimic running the whole process and check if everything works ok.

Reply:
ok